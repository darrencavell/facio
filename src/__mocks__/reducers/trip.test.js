import reducer , { initialState } from '../../reducers/trip.js'

describe('reducer trip redux', () => {
  it('should return initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot()
  })
  it('should handle TRIP_LIST', () => {
    expect(
      reducer(initialState,
      {
        type: 'TRIP_LIST',
        payload: {
          data: {
            "start_station": "Surabaya",
            "end_station": "Semarang",
            "eta": "2020-08-20 00:00:00",
            "status": "Transit",
            "driver_name": "John Doe",
            "created_at": "2020-08-18 17:12:00",
            "_id": "5f521f25d899cd03e8b4de79"
          }
        }
      })
    ).toMatchSnapshot()
  })
  it('should handle TRIP_LIST_LOADING', () => {
    expect(
      reducer(initialState,
      {
        type: 'TRIP_LIST',
        payload: true
      })
    ).toMatchSnapshot()
  })

  it('should handle ADD_TRIP', () => {
    expect(
      reducer(initialState,
      {
        type: 'ADD_TRIP',
        payload: {
          data: {
            "start_station": "Surabaya",
            "end_station": "Semarang",
            "eta": "2020-08-20 00:00:00",
            "status": "Transit",
            "driver_name": "John Doe",
            "created_at": "2020-08-18 17:12:00"
          }
        }
      })
    ).toMatchSnapshot()
  })
  
  it('should handle TRIP_ADD_LOADING', () => {
    expect(
      reducer(initialState,
      {
        type: 'TRIP_ADD_LOADING',
        payload: true
      })
    ).toMatchSnapshot()
  })
})