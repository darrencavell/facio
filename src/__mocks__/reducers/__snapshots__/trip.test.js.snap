// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action redux should handle ADD_TRIP 1`] = `
Object {
  "add": Object {
    "response": Object {
      "data": Object {
        "created_at": "2020-08-18 17:12:00",
        "driver_name": "John Doe",
        "end_station": "Semarang",
        "eta": "2020-08-20 00:00:00",
        "start_station": "Surabaya",
        "status": "Transit",
      },
    },
  },
  "lists": Object {},
}
`;

exports[`action redux should handle ADD_TRIP_LOADING 1`] = `
Object {
  "add": Object {
    "loading": true,
  },
  "lists": Object {},
}
`;

exports[`action redux should handle TRIP_LIST 1`] = `
Object {
  "add": Object {},
  "lists": Object {
    "response": Object {
      "_id": "5f521f25d899cd03e8b4de79",
      "created_at": "2020-08-18 17:12:00",
      "driver_name": "John Doe",
      "end_station": "Semarang",
      "eta": "2020-08-20 00:00:00",
      "start_station": "Surabaya",
      "status": "Transit",
    },
  },
}
`;

exports[`action redux should handle TRIP_LIST_LOADING 1`] = `
Object {
  "add": Object {},
  "lists": Object {
    "response": undefined,
  },
}
`;

exports[`action redux should return initial state 1`] = `
Object {
  "add": Object {},
  "lists": Object {},
}
`;

exports[`action trip redux should handle ADD_TRIP 1`] = `
Object {
  "add": Object {
    "response": Object {
      "data": Object {
        "created_at": "2020-08-18 17:12:00",
        "driver_name": "John Doe",
        "end_station": "Semarang",
        "eta": "2020-08-20 00:00:00",
        "start_station": "Surabaya",
        "status": "Transit",
      },
    },
  },
  "lists": Object {},
}
`;

exports[`action trip redux should handle ADD_TRIP_LOADING 1`] = `
Object {
  "add": Object {
    "loading": true,
  },
  "lists": Object {},
}
`;

exports[`action trip redux should handle TRIP_LIST 1`] = `
Object {
  "add": Object {},
  "lists": Object {
    "response": Object {
      "_id": "5f521f25d899cd03e8b4de79",
      "created_at": "2020-08-18 17:12:00",
      "driver_name": "John Doe",
      "end_station": "Semarang",
      "eta": "2020-08-20 00:00:00",
      "start_station": "Surabaya",
      "status": "Transit",
    },
  },
}
`;

exports[`action trip redux should handle TRIP_LIST_LOADING 1`] = `
Object {
  "add": Object {},
  "lists": Object {
    "response": undefined,
  },
}
`;

exports[`action trip redux should return initial state 1`] = `
Object {
  "add": Object {},
  "lists": Object {},
}
`;

exports[`reducer trip redux should handle ADD_TRIP 1`] = `
Object {
  "add": Object {
    "response": Object {
      "data": Object {
        "created_at": "2020-08-18 17:12:00",
        "driver_name": "John Doe",
        "end_station": "Semarang",
        "eta": "2020-08-20 00:00:00",
        "start_station": "Surabaya",
        "status": "Transit",
      },
    },
  },
  "lists": Object {},
}
`;

exports[`reducer trip redux should handle ADD_TRIP_LOADING 1`] = `
Object {
  "add": Object {
    "loading": true,
  },
  "lists": Object {},
}
`;

exports[`reducer trip redux should handle TRIP_ADD_LOADING 1`] = `
Object {
  "add": Object {
    "loading": true,
  },
  "lists": Object {},
}
`;

exports[`reducer trip redux should handle TRIP_LIST 1`] = `
Object {
  "add": Object {},
  "lists": Object {
    "response": Object {
      "_id": "5f521f25d899cd03e8b4de79",
      "created_at": "2020-08-18 17:12:00",
      "driver_name": "John Doe",
      "end_station": "Semarang",
      "eta": "2020-08-20 00:00:00",
      "start_station": "Surabaya",
      "status": "Transit",
    },
  },
}
`;

exports[`reducer trip redux should handle TRIP_LIST_LOADING 1`] = `
Object {
  "add": Object {},
  "lists": Object {
    "response": undefined,
  },
}
`;

exports[`reducer trip redux should return initial state 1`] = `
Object {
  "add": Object {},
  "lists": Object {},
}
`;
